sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/implementation/AccessTokenCacheImpl.java

```diff
@@ -21,64 +23,63 @@
+     * @param tokenCredential the credential to be used to acquire token from.
      */
-    public AccessTokenCache(Supplier<Mono<AccessToken>> tokenSupplier) {
-        Objects.requireNonNull(tokenSupplier, "The token supplier cannot be null");
+    public AccessTokenCacheImpl(TokenCredential tokenCredential) {
+        Objects.requireNonNull(tokenCredential, "The token credential cannot be null");
         this.wip = new AtomicReference<>();
-        this.tokenSupplier = tokenSupplier;
+        this.tokenCredential = tokenCredential;
         this.shouldRefresh = accessToken -> OffsetDateTime.now()
             .isAfter(accessToken.getExpiresAt().minus(REFRESH_OFFSET));
     }
 
-    /**
-     * Asynchronously get a token from either the cache or replenish the cache with a new token.
-     * @return a Publisher that emits an AccessToken
-     */
-    public Mono<AccessToken> getToken() {
-        return getToken(this.tokenSupplier, false);
-    }
-
     /**
      * Asynchronously get a token from either the cache or replenish the cache with a new token.
      *
-     * @param tokenSupplier The method to get a new token
-     * @param forceRefresh The flag indicating if the cache needs to be skipped and a token needs to be fetched via the
-     * credential.
+     * @param tokenRequestContext The request context for token acquisition.
      * @return The Publisher that emits an AccessToken
      */
-    public Mono<AccessToken> getToken(Supplier<Mono<AccessToken>> tokenSupplier, boolean forceRefresh) {
-        return Mono.defer(retrieveToken(tokenSupplier, forceRefresh))
+    public Mono<AccessToken> getToken(TokenRequestContext tokenRequestContext) {
+        return Mono.defer(retrieveToken(tokenRequestContext))
             // Keep resubscribing as long as Mono.defer [token acquisition] emits empty().
             .repeatWhenEmpty((Flux<Long> longFlux) -> longFlux.concatMap(ignored -> Flux.just(true)));
     }
 
-    private Supplier<Mono<? extends AccessToken>> retrieveToken(Supplier<Mono<AccessToken>> tokenSupplier,
-                                                                boolean forceRefresh) {
+    private Supplier<Mono<? extends AccessToken>> retrieveToken(TokenRequestContext tokenRequestContext) {
         return () -> {
             try {
                 if (wip.compareAndSet(null, MonoProcessor.create())) {
                     final MonoProcessor<AccessToken> monoProcessor = wip.get();
                     OffsetDateTime now = OffsetDateTime.now();
                     Mono<AccessToken> tokenRefresh;
                     Mono<AccessToken> fallback;
+
+                    Supplier<Mono<AccessToken>> tokenSupplier = () ->
+                        tokenCredential.getToken(this.tokenRequestContext);
+
+                    boolean forceRefresh = !(this.tokenRequestContext != null
+                        && this.tokenRequestContext.getClaims().equals(tokenRequestContext.getClaims())
```

Comment:
I don't believe you should always force refresh if the cached `tokenRequestContext.getClaims()` doesn't match the requested claims. If the requested claims in null it doesn't matter what the claims the cached context had requested.